src/sqlfluff/cli/commands.py

```diff
@@ -788,25 +802,14 @@ def _paths_fix(
     # NB: We filter to linting violations here, because they're
     # the only ones which can be potentially fixed.
     num_fixable = result.num_violations(types=SQLLintError, fixable=True)
+
     if num_fixable > 0:
-        if formatter.verbosity >= 0:
+        if not force and formatter.verbosity >= 0:
             click.echo("==== fixing violations ====")
+
         click.echo(f"{num_fixable} " "fixable linting violations found")
-        if force:
-            if warn_force and formatter.verbosity >= 0:
-                click.echo(
-                    f"{formatter.colorize('FORCE MODE', Color.red)}: "
-                    "Attempting fixes..."
-                )
-            success = do_fixes(
-                result,
-                formatter,
-                types=SQLLintError,
-                fixed_file_suffix=fixed_suffix,
-            )
-            if not success:
-                sys.exit(EXIT_FAIL)  # pragma: no cover
-        else:
```

Comment:
This has been moved up to happen _during the linting step_ above and so doesn't need to happen here anymore.